libraries/AP_Compass/AP_Compass.h

```diff
@@ -333,10 +336,20 @@ friend class AP_Compass_Backend;
 
 private:
     static Compass *_singleton;
+
+    // Use Priority and StateIndex typesafe index types 
+    // to distinguish between two different type of indexing
+    // We use StateIndex for access by Backend
+    // and Priority for access by Frontend
+    DECLARE_TYPESAFE_INDEX(Priority, int);
+    DECLARE_TYPESAFE_INDEX(StateIndex, int);
+
     /// Register a new compas driver, allocating an instance number
     ///
-    /// @return number of compass instances
-    uint8_t register_compass(void);
+    /// @param  dev_id                   Dev ID of compass to register against
+    ///
+    /// @return instance number saved against the dev id or first available empty instance number
+    bool register_compass(int32_t dev_id, uint8_t& instance);
```

Comment:
is the register_compass's first argument really a device id or a bus_id?  should it be a uint32_t instead of an int32_t?

Reply:
We store this parameter as an int32_t, so its right to be consistent across the board.